platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/files/MemoryIndexDiskKeyValueStore.java

```diff
@@ -89,23 +73,37 @@ public class MemoryIndexDiskKeyValueStore<D> implements AutoCloseable, Snapshota
      */
     private final AtomicLong maxValidKey;
 
+    /**
+     * A function that will be called to report the duration of the compaction, in ms
+     */
+    private final BiConsumer<Integer, Long> reportDurationMetricFunction;
+    /**
+     * A function that will be called to report the amount of space saved by the compaction, in Mb
+     */
+    private final BiConsumer<Integer, Double> reportSavedSpaceMetricFunction;
+
+    private final Runnable updateStatsFunction;
+
     /**
      * Construct a new MemoryIndexDiskKeyValueStore
      *
-     * @param storeDir The directory to store data files in
-     * @param storeName The name for the data store, this allows more than one data store in a
-     *     single directory.
-     * @param legacyStoreName Base name for the data store. If not null, the store will process
-     *     files with this prefix at startup. New files in the store will be prefixed with {@code
-     *     storeName}
-     * @param dataItemSerializer Serializer for converting raw data to/from data items
-     * @param loadedDataCallback call back for handing loaded data from existing files on startup.
-     *     Can be null if not needed.
-     * @param keyToDiskLocationIndex The index to use for keys to disk locations. Having this passed
-     *     in allows multiple MemoryIndexDiskKeyValueStore stores to share the same index if there
-     *     key ranges do not overlap. For example with internal node and leaf paths in a virtual map
-     *     tree. It also lets the caller decide the LongList implementation to use. This does mean
-     *     the caller is responsible for snapshot of the index.
+     * @param storeDir                       The directory to store data files in
```

Comment:
There is no requirement to align comments like this, and it doesn't really improve readability (particularly as the rendered view is readily available).

Reply:
I didn't do the initial formatting but I reformatted it as you asked.

Reply:
I wasn't suggesting remove all formatting (spotless will ask for line wrapping, for instance).  More suggesting that there wasn't a need to reformat as you had with everything aligned way to the right.
It's hard to tell what changed in a comment block if the whole block is reformatted.

Reply:
I didn't to it explicitly. It either was there before or was introduced by IDE.